include/vctr/Containers/VctrBase.h

```diff
@@ -390,18 +394,55 @@ class VctrBase : private Config,
     }
 
     /** Returns the number of elements that are equal to valueToLookFor. */
-    constexpr size_t count (const ElementType& valueToLookFor)
+    constexpr size_t count (const ElementType& valueToLookFor) const
     {
         return std::count (begin(), end(), valueToLookFor);
     }
 
     /** Returns the number of elements that satisfy predicate. */
-    template <is::elementPredicateFunction<ElementType> Fn>
-    constexpr size_t countIf (Fn&& predicate)
+    template <is::functionWithSignatureOrImplicitlyConvertible<bool (const ElementType&)> Fn>
+    constexpr size_t countIf (Fn&& predicate) const
     {
         return std::count_if (begin(), end(), std::forward<Fn> (predicate));
     }
 
+    /** Returns true if all elements satisfy the predicate or if it is empty */
+    template <is::functionWithSignatureOrImplicitlyConvertible<bool (const ElementType&)> Fn>
+    constexpr bool all (Fn&& predicate) const
+    {
+        return std::all_of (begin(), end(), std::forward<Fn> (predicate));
+    }
+
+    /** Returns true if one or more elements satisfy the predicate */
+    template <is::functionWithSignatureOrImplicitlyConvertible<bool (const ElementType&)> Fn>
+    constexpr bool any (Fn&& predicate) const
+    {
+        return std::any_of (begin(), end(), std::forward<Fn> (predicate));
+    }
+
+    /** Returns true if all elements are equal to value or if it is empty*/
+    template <std::equality_comparable_with<ElementType> T>
+    constexpr bool allElementsEqual (const T& value) const
+    {
+        return all ([&] (const auto& e) { return e == value; });
+    }
+
+    /** Returns true if all elements are equal to themselves */
+    constexpr bool allElementsEqual() const
```

Comment:
Would it be thinkable that this is used with types that do not have an equality operator?

Reply:
A lot is possible, I'll add an according constraint here